pkg/handler/processor/guesser/type_cyclonedx.go

```diff
@@ -38,6 +38,9 @@ func (_ *cycloneDXTypeGuesser) GuessDocumentType(blob []byte, format processor.F
 		decoder := cdx.NewBOMDecoder(reader, cdx.BOMFileFormatJSON)
 		err := decoder.Decode(bom)
 		if err == nil && bom.BOMFormat == cycloneDXFormat {
+			if bom.Vulnerabilities != nil {
+				return processor.DocumentCdxVex
+			}
```

Comment:
QQ: would CDX Vex files be only VEX? Or can 1 document be both an SBOM and a VEX file?

Reply:
I think it could be both from the looks of the attributes listed [here](https://cyclonedx.org/docs/1.5/json/#vulnerabilities_items_properties_items_value)

Reply:
Based on Brandon's comment, if it can be both. We should start with the CDX SBOM parser and call CDX VEX if needed from the CDX SBOM parser